autometa/validation/datasets.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+COPYRIGHT
+Copyright 2021 Ian J. Miller, Evan R. Rees, Kyle Wolf, Siddharth Uppal,
+Shaurya Chanana, Izaak Miller, Jason C. Kwan
+
+This file is part of Autometa.
+
+Autometa is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+COPYRIGHT
+
+pulling data from google drive dataset with simulated or synthetic communities
+"""
+
+
+import gdown
+import os
+import sys
+import logging
+import pandas as pd
+
+from autometa.common.utilities import internet_is_connected
+
+logger = logging.getLogger(__name__)
+
+
+def download(
+    community_type: str, community_size: list, file_names: list, dir_path: str
+) -> None:
+
+    """Downloads the files specified in a dictionary.
+
+    Parameters
+    ----------
+    community_type : str
+        specifies the type of dataset that the user would like to download from
+    community_size : list
+        specifies the size of dataset that the user would like to download
+    file_names : list
+        specifies the file(s) that the user would like to download
+    dir_path : str
+        dir_path_size path where the user wants to download the file(s)
+
+    Returns
+    -------
+    None
+        download is completed through gdown
+
+    """
+
+    if community_type == "synthetic" or community_type == "all":
+        sys.exit("Haven't implemented that yet")
+
+    for each_community_size in community_size:
+        df = pd.read_csv("gdown_fileIDs.csv", dtype=str)
+        df = df.query(f'dataset == "{each_community_size}"')
+        dir_path_size = os.path.join(dir_path, each_community_size)
+        # make a new directory
+        if not os.path.exists(dir_path_size):
+            os.mkdir(dir_path_size)
+
+        for file_name in file_names:
+            file_id = df.query(f'file == "{file_name}"')["file_id"].to_list()[0]
+            dir_path_final = os.path.join(dir_path_size, file_name)
+            url = f"https://drive.google.com/uc?id={file_id}"
+
+            gdown.download(url, dir_path_final)
```

Comment:
`dir_path_size` and `dir_path_final` correspond to the `community_size` output directory and the `community_size` file, respectively. 

- (minor nitpick) Could you rename these variables to represent the directory and file?

```suggestion
        community_size_outdir = os.path.join(dir_path, each_community_size)
        # make a new directory
        if not os.path.exists(community_size_outdir):
            os.makedirs(community_size_outdir)

        for file_name in file_names:
            file_id = df.set_index(["dataset", "file"]).loc[(community_size, file_name), 'file_id']
            community_size_filepath = os.path.join(community_size_outdir, file_name)
            url = f"https://drive.google.com/uc?id={file_id}"

            gdown.download(url, community_size_filepath)
```

- (minor nitpick) Similar to the query you performed above, could the quoting usage be reduced?

I've noticed when I have to look-up a particular value in pandas, indexing is one of the first things that comes to mind. Indexing can be quite powerful to access a particular value for multiple columns.

## Example:

```python
# hierarchical index when reading in the table
df = pd.read_csv("gdown_fileIDs.csv", dtype=str, index_col=["dataset", "file"])
```

### Setting the index after we've read in the table.

```python
## Two methods: 
### 1. in place, no variable re-assignment (will return NoneType)
df.set_index(['dataset', 'file'], inplace=True)
### 2. with variable re-assignment (returns the dataframe with the set index)
df = df.set_index(['dataset', 'file'])
```

### Looking up a value with the hierarchical index

```python
# Make sure hierarchical index is set
df = df.set_index(['dataset', 'file'])
community_size = '78Mbp'
file_name = 'orfs.faa.gz'
df.loc[(community_size, file_name), 'file_id']
# The same can all be done without variable reassignment or having a new df with the set index.
### input
df.set_index(['dataset', 'file']).loc[('78Mbp', 'orfs.faa.gz'), 'file_id']
### output
'1z-VveBEeVRyr7eushHvHCrdqDShNkaBq'
```

### Looking up all values with hierarchical index

Look-up for a generic pandas Dataframe usage: `df.loc[index, cols]`

```python
# Make sure hierarchical index is set
df = df.set_index(['dataset', 'file'])
community_size = '78Mbp'
file_name = 'orfs.faa.gz'
# In
df.loc[(community_size, file_name)] # Note there is no ',' after the 2-tuple.
# Out
community                            simulated
file_id      1z-VveBEeVRyr7eushHvHCrdqDShNkaBq
Name: (78Mbp, orfs.faa.gz), dtype: object

## NOTE: the columns can be accessed via there namespace with this as well
df.loc[(community_size, file_name)].file_id
df.loc[(community_size, file_name)].community
```

## Original

```python
file_id = df.query(f'file == "{file_name}"')["file_id"].to_list()[0]
````
## Other suggestions using index

```python
df = df.set_index(['community', 'file'])
# Access within loc (IMO preferred)
file_id = df.loc[(community_size, file_name), "file_id"]
# Access via namespace
file_id = df.loc[(community_size, file_name)].file_id
```

Reply:
Fantastic! This is so much more elegant than how I was doing it. Thanks for the guidance!